cityofaustin/atd-moped
# Show features on the map with basic symbology

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/9927

This one adds project component features to the map shown with the same line and point styles used in the components map. Instead of blue points and lines, the features use the color of the current status of their related project.

Feedback that I need 🙏:
- Projects with status of "On Hold" show in light grey which is very hard to see on both street and satellite base maps. I've added a small outline to fight this issue but it doesn't feel quite right. I'm opposed to changing the colors since they are very ingrained in this app and communicate the project status all over the place, but I'd like to hear if anyone has ideas about how to make this better. I'm going to do some reading about it.

Collecting some map style resources here:
- [ColorBrewer](https://colorbrewer2.org/#type=sequential&scheme=BuGn&n=3)
- [ESRI: 6 Easy Ways to Improve Your Maps (mentions outlines)](https://www.esri.com/arcgis-blog/products/arcgis-online/mapping/6-easy-ways-to-improve-your-maps/?rmedium=redirect&rsource=blogs.esri.com/esri/arcgis/2016/04/12/six_easy_ways_to_improve_your_maps
)
- [The what, when, and how to symbolize by size (mostly about symbol size but touches on transparency and outlines at the end)](https://www.esri.com/arcgis-blog/products/arcgis-online/mapping/size/)
## Testing
**URL to test:** 

Local

**Steps to test:**
1. Go to the project list view and toggle the **Show map** switch on
2. You should see the spinner in the search bar start spinning now that the project features are fetching
3. You should see the map load all of the components from all projects in the app on the map in the colors used for the project status badges
4. Click the advanced filters button and add a filter for **Full name contains cameron**
5. You should see features with various statuses in the NE area of Austin
6. Add another filter for **Status is potential** and you should only see orange features on the map
7. Open the project list view advanced filter dropdown and notice that the width of the dropdown matches the width of the search box above from large to mobile screen widths
8. Close the dropdown and watch the responsive behavior of the map switch, download button, and the search button. The search button only appears on mobile screens. thx @johnclary for this feedback in the last pr 🙏
9. Check places where the status badge renders like the project list view, dashboard tables, project summary view, and project map tab header to make sure that the code edits in `ProjectStatusBadge.js` didn't disturb the way they render.

---
#### Ship list
